tests/v2/validation/configmaps/generated_fields_test.go

```diff
@@ -0,0 +1,104 @@
+package configmaps
+
+import (
+	"fmt"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	steveV1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	cm "github.com/rancher/rancher/tests/framework/extensions/configmaps"
+	"github.com/rancher/rancher/tests/framework/extensions/kubeapi/configmaps"
+	"github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/api/core/v1"
+	"strings"
+	"testing"
+)
+
+const (
+	APIVersion         = "v1"
+	Kind               = "ConfigMap"
+	configMapsEndpoint = "configmaps"
+	configMapNamespace = "default"
+)
+
+type ConfigMapTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	steveClient        *steveV1.Client
+	session            *session.Session
+	cluster            *management.Cluster
+	nameSpacedV1Client *steveV1.NamespacedSteveClient
+	configMapPayload   *cm.GenFieldTest
+}
+
+func (c *ConfigMapTestSuite) TearDownSuite() {
+	c.session.Cleanup()
+}
+
+func (c *ConfigMapTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	c.session = testSession
+	client, err := rancher.NewClient("", c.session)
+	require.NoError(c.T(), err)
+	c.client = client
+	c.steveClient = client.Steve
+}
+
+func (c *ConfigMapTestSuite) TestSteveGeneratedFields() {
+
+	configData := map[string]string{
+		"env": "qa",
+	}
+	labels, annotations := map[string]string{}, map[string]string{}
+	configMapName := namegenerator.AppendRandomString("test-configmap")
+	configmap, err := configmaps.CreateConfigMap(c.client, c.client.RancherConfig.ClusterName, configMapName, "auto-generated", configMapNamespace, configData, labels, annotations)
+	require.NoError(c.T(), err)
+
+	logrus.Infof("created a configmap(%v, configmap).............", configmap.Name)
+
+	c.configMapPayload = &cm.GenFieldTest{}
+	copyConfigMapToPayload(configmap, c.configMapPayload)
+
+	// Update the configmap payload with the fields that we want to test.
+	c.configMapPayload.Metadata.Fields = "test-fields"
+	c.configMapPayload.Metadata.Relationships = "test-relationships"
+	c.configMapPayload.Metadata.State = "test-state"
+	c.configMapPayload.Data["foo"] = "bar"
+
+	steveClient := c.client.Steve
+	headers, _, err := steveClient.SteveType("namespace").NamespacedSteveClient(configMapNamespace).PerformPutCaptureHeaders(c.client.RancherConfig.Host, c.client.RancherConfig.AdminToken, configMapsEndpoint, configmap.Name, c.configMapPayload)
```

Comment:
also Andrew, if this for a config map why is the Steve type namespace?

Reply:
Ah, just an accident. Fixed! Thank you! 😁